### PR TITLE
feat: enable download of go client for host arch

### PR DESCRIPTION
**Pull request by nestoracunablanco**
This commit is a step towards enabling builds for different architectures.

**What this PR does / why we need it**: s390x enablement

**Special notes for your reviewer**:

**Release note**:

```release-note NONE

```

### CODE REVIEW

**Comment by nestoracunablanco** [comment]
@qinqon @RamLavi what do you think about this change?

File: hack/install-go.sh [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 destination=$1
 version=$(curl -s https://go.dev/dl/?mode=json | jq -r ".[0].version")
-tarball=$version.linux-amd64.tar.gz
+arch=$(uname -m | sed 's/x86_64/amd64/')
```
Comment (RamLavi) [inline hack/install-go.sh:5]: I'm not sure I understand. 
If the machine running is `x86_64` then why would I want to substitute it to `amd64`?
Shouldn't it just be something like
```
arch=$(uname -m)
if [ "$arch" != "x86_64" ] && [ "$arch" != "amd64" ]; then
  echo "Unsupported architecture: $arch"
  exit 1
fi
```
?

Comment (nestoracunablanco) [reply hack/install-go.sh:5]: x86_64" and "amd64" are both valid terms used to refer to the same architecture. However, when you check the architecture via the command line using uname -m, it returns "x86_64". The Go programming language project, on the other hand, prefers the term "amd64." Therefore, if we are working with the amd64 architecture, we need to rename it accordingly.

Comment (RamLavi) [reply hack/install-go.sh:5]: ah I see. Learned something new :) 
Please add this explanation to the commit desc.

Comment (nestoracunablanco) [reply hack/install-go.sh:5]: @RamLavi done. Thanks for the feedback.

**Pull request merged**